flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/serialization/SpanningRecordSerializer.java

```diff
@@ -127,6 +116,7 @@ public void reset() {
 
 	@Override
 	public void prune() {
+		serializationBuffer.clear();
```

Comment:
Although I can not understand why it would affect the performance in theory, I agree with your point from the consistency of view  to keep this logic.

Maybe it seems better set this position to 0 inside the `serializationBuffer.pruneBuffer`. It's up to you for the decision. :)